tests/integration/helpers.bash

```diff
@@ -200,75 +207,85 @@ function fail() {
 # support it, the test is skipped with a message.
 function requires() {
 	for var in "$@"; do
+		skip=
 		case $var in
 		criu)
 			if [ ! -e "$CRIU" ]; then
-				skip "test requires ${var}"
+				skip=1
 			fi
 			;;
 		root)
 			if [ "$ROOTLESS" -ne 0 ]; then
-				skip "test requires ${var}"
+				skip=1
 			fi
 			;;
 		rootless)
 			if [ "$ROOTLESS" -eq 0 ]; then
-				skip "test requires ${var}"
+				skip=1
 			fi
 			;;
 		rootless_idmap)
 			if [[ "$ROOTLESS_FEATURES" != *"idmap"* ]]; then
-				skip "test requires ${var}"
+				skip=1
 			fi
 			;;
 		rootless_cgroup)
 			if [[ "$ROOTLESS_FEATURES" != *"cgroup"* ]]; then
-				skip "test requires ${var}"
+				skip=1
 			fi
 			;;
 		rootless_no_cgroup)
 			if [[ "$ROOTLESS_FEATURES" == *"cgroup"* ]]; then
-				skip "test requires ${var}"
+				skip=1
+			fi
+			;;
+		cgroups_freezer)
+			init_cgroup_paths
+			if [[ "$CGROUP_SUBSYSTEMS" != *"freezer"* ]]; then
+				skip=1
 			fi
 			;;
 		cgroups_kmem)
 			init_cgroup_paths
 			if [ ! -e "${CGROUP_MEMORY_BASE_PATH}/memory.kmem.limit_in_bytes" ]; then
-				skip "Test requires ${var}"
+				skip=1
 			fi
 			;;
 		cgroups_rt)
 			init_cgroup_paths
 			if [ ! -e "${CGROUP_CPU_BASE_PATH}/cpu.rt_period_us" ]; then
-				skip "Test requires ${var}"
+				skip=1
 			fi
 			;;
 		cgroups_v1)
 			init_cgroup_paths
 			if [ "$CGROUP_UNIFIED" != "no" ]; then
-				skip "Test requires cgroups v1"
+				skip=1
 			fi
 			;;
 		cgroups_v2)
 			init_cgroup_paths
 			if [ "$CGROUP_UNIFIED" != "yes" ]; then
-				skip "Test requires cgroups v2 (unified)"
+				skip=1
 			fi
 			;;
 		systemd)
 			if [ -z "${RUNC_USE_SYSTEMD}" ]; then
-				skip "Test requires systemd"
+				skip=1
 			fi
 			;;
 		no_systemd)
 			if [ -n "${RUNC_USE_SYSTEMD}" ]; then
-				skip "Test requires no systemd"
+				skip=1
 			fi
 			;;
 		*)
-			fail "BUG: Invalid requires ${var}."
+			fail "BUG: Invalid requires $var."
 			;;
 		esac
+		if [ -n "$skip" ]; then
+			skip "test requires $var"
```

Comment:
nit: Can we avoid using the same identifier for the variable and the function. (Surprised to know they don't conflict)

Reply:
Sure. The reason they don't conflict is that variables and functions are in different namespaces.

Reply:
Done.